libs/@guardian/libs/src/storage/storage.ts

```diff
@@ -98,6 +98,38 @@ class StorageFactory {
 	setRaw(key: string, value: string): void {
 		return this.#storage?.setItem(key, value);
 	}
+
+	/**
+	 * Get all of the keys in storage.
+	 *
+	 * @returns an array of all keys in the storage object,
+	 * or undefined if storage is not available.
+	 *
+	 * @note This is not a method in the native API, and gets
+	 * all keys by iterating over the entire storage object.
+	 * It is therefore potentially subject to race conditions
+	 * and if other code is modifying storage at the same time
+	 * then the results may be inaccurate.
+	 *
+	 * @note This method depends on the `key` method on the native API,
+	 * which gets keys by index. But the _order_ of keys in storage is
+	 * dependent on user-agent implementation, and so you should not rely
+	 * on the order of keys returned by this method. (See
+	 * https://developer.mozilla.org/en-US/docs/Web/API/Storage/key for more.)
+	 */
+	keys(): string[] | undefined {
```

Comment:
couple of thoughts:

1. how do you feel about something more explicit, like `allKeys()`? 
2. should we also add `key()` to our api? it only exists to discourage footguns of using the native one. forcing people to use the native one (if they need `key()`) seems wrong